File: signals/sender.c
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>
#include <unistd.h>

int main (int argc, char *argv[])
{
    if (argc != 2)
    {
        printf("Informe o PID do processo Receiver\n");
        exit(1);
    }

    // obtendo o PID para mandar o sinal
    int PID = atoi(argv[1]);

    // envia o SIGUSR1 a cada 5 segundos
    kill(PID, SIGUSR1);

    return 0;
}

File: ipc/signals/python/sender.py
import os
import signal
import sys

if (len(sys.argv) != 2):
    print("Informe o PID do processo Receiver!")
    exit(1)


pid = int(sys.argv[1])


# Enviando o sinal para o processo 
os.kill(pid, signal.SIGUSR1)

